Makefile

```
@@ -786,13 +820,20 @@ clean-tesseract:
 # install git and parallel first (which is required for the module updates)
 deps-ubuntu:
 	apt-get -y install git parallel
+ifneq ($(suffix $(PYTHON)),)
+# install specific Python version in system via PPA
+	apt-get install -y software-properties-common
+	add-apt-repository -y ppa:deadsnakes/ppa
+	apt-get update
+	apt-get install -y --no-install-recommends $(notdir $(PYTHON))-dev $(notdir $(PYTHON))-venv
+endif
```

Why was this block added? I don't think that it is a good idea to install a Python package from PPA just because the user added `PYTHON=python3.7` to the build options. It can either overwrite an existing installation or install an unneeded 2nd Python installation.

I noticed these instructions while building a Docker image with GitLab actions which already provides any desired Python version.

---

Because our current reference platform, i.e. the one targeted by `deps-ubuntu`, is still Ubuntu 18.04. And that uses 3.6 by default for python3, and needs PPAs for more recent versions. (Same with our ocrd_tesserocr recipe.)

I agree it's not a good pattern, but we need some kind of dual version support during the grace period between Ubuntu 18 and 20 (Python 3.6 and 3.8). Think of it as something we can drop as soon as the shift to 20 is complete. (And after that we should find better mechanisms to encapsulate system dependencies.)

I think the natural way to avoid interference with tailored environments like CI is to _not_ set `PYTHON` there, but instead make it that environment's default version.

---

18.04 does not need PPA for Python 3.7: https://packages.ubuntu.com/search?keywords=python3.7. And I don't think that the installation of Python should be part of the Makefile rules. Let it fail if someone says `make all PYTHON=python3.7` without providing that version.